### PR TITLE
arm64: dts: qcom: msm8916-samsung-j5nlte: add sound

### PR DESCRIPTION
**Pull request by nergzd723**
Works:

- Speaker
- Earphones
- Microphone
- Earpiece

Jack detection works with jack-gpios hack.

### CODE REVIEW

**Comment by Minecrell** [comment]
> Jack detection doesn't work for some reason, any ideas?

You can try making an extra commit like this one (note: it's really a bad hack but usually works somewhat): https://github.com/msm8916-mainline/linux/commit/85ad9b900e03f3766b10eb898b8bb70205be0295

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 1]
```diff
@@ -24,6 +25,31 @@
...
+
+		wcnss_mem: wcnss@8be00000 {
+			reg = <0x0 0x8be00000 0x0 0x600000>;
+			no-map;
+		};
+
+		venus_mem: venus@8c400000 {
+			reg = <0x0 0x8c400000 0x0 0x600000>;
+			no-map;
+		};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:52]: This doesn't belong into this PR (probably belongs into the modem one)

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 2]
```diff
@@ -147,12 +177,54 @@
...
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&cdc_pdm_lines_act>;
+	pinctrl-1 = <&cdc_pdm_lines_sus>;
+
+	model = "msm8916";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "Headset Mic",
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:190]: I don't know what Headset Mic is but it shouldn't be here :S 

This seems to be some real mess downstream... I'm not entirely sure what it means, have you tried connecting a headset with microphone and recording audio from it?

This should be either `MIC BIAS External2` or `MIC BIAS Internal2`...

**Comment by Minecrell** [comment]
@jugendhacker Do you happen to have a headset with microphone you could use to test this on j5nlte (also see comment above)? Apparently @nergzd723 can't find one.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts (hunk 2 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:190]: `Headset Mic`, `MIC BIAS External2` and `MIC BIAS Internal2` are all working.

**Comment by stephan-gh** [comment]
Looks like this is being continued in https://github.com/msm8916-mainline/linux/pull/237

**Pull request closed without merging**